app/script/event/EventService.js

```diff
@@ -79,32 +79,19 @@ z.event.EventService = class EventService {
    *  until either limit or lower bound is reached.
    *
    * @param {string} conversationId - ID of conversation
-   * @param {Date} [lowerBound=new Date(0)] - Load from this date (included)
-   * @param {Date} [upperBound=new Date()] - Load until this date (excluded)
+   * @param {Date} [fromDate=new Date(0)] - Load from this date (included)
+   * @param {Date} [toDate=new Date()] - Load until this date (excluded)
    * @param {number} [limit=Number.MAX_SAFE_INTEGER] - Amount of events to load
    * @returns {Promise} Resolves with the retrieved records
    */
-  loadPrecedingEvents(
-    conversationId,
-    lowerBound = new Date(0),
-    upperBound = new Date(),
-    limit = Number.MAX_SAFE_INTEGER
-  ) {
-    if (!_.isDate(lowerBound) || !_.isDate(upperBound)) {
-      const errorMessage = `Lower bound (${typeof lowerBound}) and upper bound (${typeof upperBound}) must be of type 'Date'.`;
-      throw new Error(errorMessage);
-    }
+  loadPrecedingEvents(conversationId, fromDate = new Date(0), toDate = new Date(), limit = Number.MAX_SAFE_INTEGER) {
+    const includeParams = {
+      includeFrom: true,
+      includeTo: false,
+    };
 
-    if (lowerBound.getTime() > upperBound.getTime()) {
-      const errorMessage = `Lower bound (${lowerBound.getTime()}) cannot be greater than upper bound (${upperBound.getTime()}).`;
-      throw new Error(errorMessage);
-    }
-
-    return this.storageService.db[this.EVENT_STORE_NAME]
-      .where('[conversation+time]')
-      .between([conversationId, lowerBound.toISOString()], [conversationId, upperBound.toISOString()], true, false)
+    return this._loadEventsInDateRange(conversationId, fromDate, toDate, limit, includeParams)
```

Comment:
Ok, this is a pretty serious refactoring (I merged the logic of 2 methods into a single one, namely `_loadEventsInDateRange`), but we can be pretty confident the behavior is the same because we have a pretty good test coverage of both those methods and they are all green :+1:

Reply:
Oh very good point, I forgot to migrate this method in the `noCompound` class!!
On it!